### PR TITLE
Give a choice for app to extend the length limitation of clientId even in mqtt v3.1 on the server side

### PR DESCRIPTION
**Pull request by skyguard1**
Motivation:

In the mqtt v3.1 protocol, the default maximum Client Identifier length is 23.However, in (https://github.com/netty/netty/issues/11114), there are many cases, the server may still receive a client ID with a length greater than 23. Perhaps should consider letting the user decide whether accept client id greater than 23 on the server side

Modification:

Give a choice for app to extend the length limitation of clientId even in mqtt v3.1 on the server side, thanks

### CODE REVIEW

**Review by normanmaurer — APPROVED** [review]

File: codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java [hunk 1]
```diff
@@ -64,13 +65,21 @@
...
     public MqttDecoder(int maxBytesInMessage) {
+        this(maxBytesInMessage, DEFAULT_MAX_CLIENT_ID_LENGTH);
+    }
+
+    public MqttDecoder(int maxBytesInMessage, int maxClientIdLength) {
         super(DecoderState.READ_FIXED_HEADER);
+        ObjectUtil.checkPositive(maxBytesInMessage, "maxBytesInMessage");
+        ObjectUtil.checkPositive(maxClientIdLength, "maxClientIdLength");
+        this.maxClientIdLength = maxClientIdLength;
         this.maxBytesInMessage = maxBytesInMessage;
```
Comment (normanmaurer) [inline codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java:83]: nit: you can merge the above lines... For example:

```
this.maxClientIdLength = ObjectUtil.checkPositive(maxClientIdLength, "maxClientIdLength");
```

Let's do this for all of these

**Review by normanmaurer — CHANGES_REQUESTED** [review]

**Review by normanmaurer — APPROVED** [review]

File: codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java (hunk 1 above)
Comment (skyguard1) [reply codec-mqtt/src/main/java/io/netty/handler/codec/mqtt/MqttDecoder.java]: I've done it

**Comment by normanmaurer** [comment]
@skyguard1 seems like this causes test failures... Can you check ?

**Comment by skyguard1** [comment]
sorry, I've done it

**Comment by normanmaurer** [comment]
Thanks a lot @skyguard1

**Comment by skyguard1** [comment]
@normanmaurer, it is my pleasure

**Pull request merged**